Make payment_secret mandatory and ASSUMED (#1)
erickcestari/bolts: 11-payment-encoding.md

    - MUST specify the most-preferred field first, followed by less-preferred fields, in order.

A reader:
  - MUST skip over unknown fields, OR an `f` field with unknown `version`, OR  `p`, `h` or

>   `n` fields that do NOT have `data_length`s of 52, 52, 52 or 53, respectively.

Reviewer: Since you removed the `s` you have to update here as well?
```suggestion
  `n` fields that do NOT have `data_length`s of 52, 52, or 53, respectively.
```

Author: Yes, good catch. I completely forgot about this. Thanks for pointing it out.

  description.
  - if a valid `n` field is provided:
    - MUST use the `n` field to validate the signature instead of performing signature recovery.
  - if a valid `s` field is not provided:
    - MUST fail the payment.
  - otherwise:

>     - MUST use the `s` field as [`payment_secret`](04-onion-routing.md#tlv_payload-payload-format)

Reviewer 2: It *might* also make sense to remove the requirement further up that incorrect-length `s` fields be skipped.  WDYT?

Author: Yes, I agree!

  description.
  - if a valid `n` field is provided:
    - MUST use the `n` field to validate the signature instead of performing signature recovery.

>   - MUST require a valid `s` field to be present
>     - MUST fail the payment if no valid `s` field is found
>     - MUST use the `s` field value as [`payment_secret`](04-onion-routing.md#tlv_payload-payload-format)

Reviewer 2: For consistency, I think we should use wording that's similar to the other requirements.  Something like:

```suggestion
  - if a valid `s` field is not provided:
    - MUST fail the payment.
  - otherwise:
    - MUST use the `s` field as [`payment_secret`](04-onion-routing.md#tlv_payload-payload-format)
```

Reviewer: Question: "MUST fail the payment if no valid `s` field is found". What would be an invalid `s` field? it should fail anyway, no? It seems like it should be "MUST fail the payment if no `s` field is found"?

Reviewer 2: An `s` field that's not `len=52` would be invalid.  Your suggested wording would mean the reader should *not* fail if such a field is found.

Reviewer: Yes, you're right, thanks
